### PR TITLE
Refactor Scc to enable xTB integration

### PR DESCRIPTION
**Pull request by aradi**
Steps to do:

- [x] Pack short gamma calculation into derived type
- [x] Pack Gamma-based electrostatics into derived type
- [x] Pack Poission-based electrostatics into derived type
- Generalize interface of Gamma- and Poisson-electrostatics into abstract type (subject for a further PR)

### CODE REVIEW

**Comment by bhourahine** [comment]
> Steps to do:

Do you have a  plan regarding the 3rd order (as that is also mixed into the scc routines at the moment)?

**Review by awvwgk — COMMENTED** [review]
From the overall look I really like the direction this patch is taking. It looks like it unentangles the pieces that kept me back with the electrostatics for xTB.

**Comment by aradi** [comment]
@awvwgk , @bhourahine: This is now ready for review and merge. I've fixed hopefully all bugs introduced during refactoring. Also buildbot runs through now. Coulomb-solver should be simplified at some point in a future PR.

**Comment by awvwgk** [comment]
You might want to rebase / merge master first to include the latest refactoring from the default branch.

**Comment by aradi** [comment]
@awvwgk Sure, done. Apparently, the macos-latest test fails with regression testing running for over an hour and not producing any logs. Any clue?

**Comment by awvwgk** [comment]
I restarted the CI, maybe it is an issue with the runner, if it doesn't go away it might be a bug only appearing on OSX.

File: prog/dftb+/lib_dftb/scc.F90 [hunk 1]
```diff
@@ -673,33 +549,29 @@ subroutine getAtomicGammaMatU(this, gammamat, U_h, species, iNeighbour, img2Cent
...
-        gammamat(iAt2f, iAt1) = gammamat(iAt2f, iAt1) - expGamma(R_ab, U_h(iSp2), U_h(iSp1))
-      end do
-    end do
 
-    do iAt1 = 1, this%nAtom
-      do iAt2 = 1, iAt1 - 1
-        gammamat(iAt2, iAt1) = gammamat(iAt1, iAt2)
+    if (this%elstatType /= elstatTypes%gammaFunc) then
+      call error("scc:getAtomicGammaMatU only works with Gamma-electrostatics")
+    end if
```
[the remark this answers is not in the document]
Comment (aradi) [reply prog/dftb+/lib_dftb/scc.F90:563]: It probably indeed makes sense to turn it into an assert, as the following code would crash anyway, if the condition is not fulfilled...

**Pull request merged**